add-permissions.js

```diff
@@ -13,6 +13,18 @@ module.exports = class AwsAddLambdaAccountPermissions {
     this.hooks = {
       'package:createDeploymentArtifacts': () => this.beforeDeploy(),
     };
+    this.serverless.configSchemaHandler.defineFunctionProperties(this.provider, {
+      properties: {
+        allowAccess: { type: 'array' },
+      },
+    });
+   this.serverless.configSchemaHandler.defineProvider(this.provider, {
+      provider: {
+        properties: {
+          access: { type: 'object' },
```

Comment:
Not sure whether we want to define the whole tree here or just the top level 🤷‍♂️
I guess, we can be simple as everything else is covered in code.